tensorflow/g3doc/get_started/os_setup.md

```diff
@@ -78,9 +78,9 @@ $ sudo pip3 install --upgrade https://storage.googleapis.com/tensorflow/mac/tens
 ```
 
 
-You can now [test your installation](#test_install).
+You can now [test your installation](#test-the-tensorflow-installation).
 
-## Virtualenv installation {#virtualenv_install}
```

Comment:
Almost that- it includes numerics as well as hyphens: [^a-z0-9\-], or [^\w\-]

This is the Python I'm using to make the header tags:
```python
def create_anchor_from_header(header, existingAnchors=None):
	# Strip white space on the left/right and make lower case
	out = header.strip().lower()

	# Replace non-alphanumeric characters, hyphens, and spaces with spaces
	out = re.sub(r"[^\w\- ]+", lambda x: " ", out)

	# Replace groups of white space with hyphens
	out = re.sub(r"\s+", lambda x: "-", out)

	# Remove any amount of hyphens at the end of the header
	out = re.sub(r"\-+$", "", out)

	if existingAnchors:
		if out in existingAnchors:
			i = 1
			while (out + "-" + str(i)) in existingAnchors and i < 100:
				i +=  1
			return out + "-" + str(i)
	
	return out
```

I've got the main changes down. I'm just going through and replacing links from other files to those old headers. Should be done very soon.